File: flare-web-client/src/flareclient/nativepane/flare-native-pane-context.js
// "Channel B" for panes, mirroring flare-client-plugin-context.js — live
// getters only, never a value snapshotted at registration time.
export function createPaneContext({ getScreenName, getAuthToken, getBaseHttpUrl, notifyAuthFailure, setVariable, fireAction }) {
  return {
    getScreenName: () => getScreenName(),
    getAuthToken: () => getAuthToken(),
    getBaseHttpUrl: () => getBaseHttpUrl(),
    notifyAuthFailure: () => notifyAuthFailure(),
    setVariable: (name, value) => setVariable(name, value),
    fireAction: (actionName, payload) => fireAction(actionName, payload)
  };
}

File: flare-android-client/flareclient/src/main/java/dev/flareframework/client/flare/nativepane/FlareNativePaneContext.java
package dev.flareframework.client.flare.nativepane;

import org.json.JSONObject;

/**
 * Runtime facts + callbacks handed to every native pane, fresh per bind.
 * Mirrors FlareClientPluginContext's "never cache, always re-read" discipline.
 */
public interface FlareNativePaneContext {
    String getScreenName();
    String getAuthToken();
    String getBaseHttpUrl();
    void notifyAuthFailure();
    void setVariable(String name, Object value);
    void fireAction(String actionName, JSONObject payload);
}